daemon/logger/awslogs/cloudwatchlogs_test.go

```diff
@@ -1651,113 +1683,3 @@ func BenchmarkUnwrapEvents(b *testing.B) {
 		assert.Check(b, is.Len(res, maximumLogEventsPerPut))
 	}
 }
-
-func TestNewAWSLogsClientCredentialEndpointDetect(t *testing.T) {
```

Comment:
@samuelkarp do you know the origin of these tests? Not so sure we can remove them.

Reply:
Tbc, I think it would be the responsible thing to do to keep these tests. I couldn't find a great way to test creds with sdkv2. Very open to suggestions.

I'm terrified that we'll merge and some bespoke feature of aws sdk will break, with pretty widespread blast-radius and slow release cadence....

Reply:
This test was to verify the functionality of `awslogs-credentials-endpoint`, which is a non-standard credential provider added to the awslogs driver that ECS and Fargate both depend on.

Reply:
We do use that functionality to support the [ECS task execution role](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_execution_IAM_role.html), I see above in the unit tests that there is a somewhat similar-looking unit test (at [line ~208](https://github.com/moby/moby/pull/44087/files#diff-a15897c1d4ace995483d1ed2e6414291d7181a9821b777b72f743189311a3fc1R208)). 

That test is also using a test server and changing the endpoint. I believe the same concept should still apply here. Essentially when we have `awslogs-credentials-endpoint` we are expecting that the awslogs client will grab credentials from this endpoint, using the "http" credentials endpoint provider. This was using the ["endpoint creds" credential provider from the v1 sdk,](https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/endpointcreds/) you can see where this was option was parsed and switched to endpointcreds [here in code](https://github.com/moby/moby/blob/a189651b4c772881ba87eec181312cd8bc20ec2e/daemon/logger/awslogs/cloudwatchlogs.go#L384-L395).

Anyways, maybe that was TMI, but from what I understand in your code you have ported to the v2 version of endpoint creds, so AFAICT this unit test should still be working? Was there something in particular you couldn't get working?

Reply:
It's a bit tough, because we can no longer reach into the client and grab the creds, like `creds, err := client.Config.Credentials.Get()`.

We could of course define a function like `getCredentialsProvider(info.Config[credentialsEndpointKey])` and write a test on `getCredentialsProvider`, but then there's no guarantee it actually gets applied to the client.

I rigged up a messy test that I think does test for the functionality we want to test for. Let me know what you think, and if you feel there's a better way to test this.